staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3605,6 +3605,26 @@ type VolumeMountStatus struct {
 	// depending on the mount result.
 	// +optional
 	RecursiveReadOnly *RecursiveReadOnlyMode `json:"recursiveReadOnly,omitempty" protobuf:"bytes,4,opt,name=recursiveReadOnly,casttype=RecursiveReadOnlyMode"`
+	// volumeStatus represents volume-type-specific status about the mounted
+	// volume.
+	VolumeStatus `json:"volumeStatus,omitempty" protobuf:"bytes,5,opt,name=volumeStatus"`
+}
+
+// VolumeStatus Represents the status of a mounted volume.
+// Exactly one of its members must be specified.
+type VolumeStatus struct {
+	// image represents an OCI object (a container image or artifact) pulled and mounted on the kubelet's host machine.
+	// +featureGate=ImageVolume
+	// +optional
+	Image *ImageVolumeStatus `json:"image,omitempty" protobuf:"bytes,1,opt,name=image"`
+}
+
+// ImageVolumeStatus represents the image-based volume status.
+type ImageVolumeStatus struct {
+	// ImageRef is the digest of the image used for this volume.
+	// It should have a value that's similar to the pod's status.containerStatuses[i].imageID.
```

Comment:
Do we really have no spec for it?

Is `""` a valid value?  Is `"x"` valid?  Is `"Four score and seven years ago, ..." valid?  I suspect you want to spec it as non-empty and limit the length to something like 128 bytes?

Reply:
historically we have issues with what is expected here: https://github.com/kubernetes/kubernetes/issues/115199

We cannot enforce specific value for this field in this KEP, but since KEP is using the same CRI APIs to pull images and check status.

I think this inconsistency can and should be fixed in future updates to ImageService CRI APIs

Reply:
> Can we just assert that it can't be empty and it must be less than some number (e.g. 256, 512, 1024) bytes?

This makes sense to me.
Hope this is fine with you as well @SergeyKanzhelev.
Done.